libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -393,3 +393,17 @@ void ConstantEvaluator::endVisit(TupleExpression const& _tuple)
 	if (!_tuple.isInlineArray() && _tuple.components().size() == 1)
 		m_values[&_tuple] = evaluate(*_tuple.components().front());
 }
+
+void ConstantEvaluator::endVisit(FunctionCall const& _functionCall)
+{
+	// try to interpret elementary type name expressions like uint(5)
+	if (auto const* expr = dynamic_cast<ElementaryTypeNameExpression const*>(&_functionCall.expression()))
+		if (auto const* type = expr->annotation().type)
+			if (auto const* typeType = dynamic_cast<TypeType const*>(type); typeType && typeType->actualType())
+			{
+				auto const args = _functionCall.arguments();
+				if (args.size() == 1)
+					if (auto const& val = evaluate(*args.front()))
+						m_values[&_functionCall] = convertType(val, *typeType->actualType());
+			}
```

Comment:
```suggestion
	if (auto const* typeType = dynamic_cast<TypeType const*>(expr->annotation().type); typeType->actualType())
	{
		auto const args = _functionCall.arguments();
		if (args.size() == 1)
			if (auto const& val = evaluate(*args.front()))
				m_values[&_functionCall] = convertType(val, *typeType->actualType());
	}
```

Reply:
@nikola-matic We'll remove this part anyways, but the additional ``typeType &&`` part is non-optional here :-).

Reply:
Won't it short circuit if the typeType is nullptr at assignment?

Reply:
That's not a thing for ``if(init; cond)`` - also couldn't really be, ``init`` is not even required to be convertible to ``bool``.